packages/authentication/ios/Plugin/Classes/Results/CheckAppTrackingTransparencyPermissionResult.swift

```diff
@@ -0,0 +1,31 @@
+import Capacitor
+import AppTrackingTransparency
+
+@objc public class CheckAppTrackingTransparencyPermissionResult: NSObject, Result {
+    let status: ATTrackingManager.AuthorizationStatus
+
+    init(_ status: ATTrackingManager.AuthorizationStatus) {
+        self.status = status
+    }
+
+    func toJSObject() -> AnyObject {
+        var result = JSObject()
+        result["status"] = convertStatusToPermissionStatus(status)
+        return result as AnyObject
+    }
+
+    private func convertStatusToPermissionStatus(_ status: ATTrackingManager.AuthorizationStatus) -> String {
+        switch status {
+        case .notDetermined:
+            return "prompt"
+        case .denied:
+            return "denied"
+        case .authorized:
+            return "granted"
+        case .restricted:
+            return "prompt-with-rationale"
```

Comment:
```suggestion
            return "restricted"
```

`prompt-with-rationale` actually means that the user has already been asked before but can be asked again. `Restricted`, on the other hand, means that the user has deactivated app tracking in the device settings and therefore authorization cannot be obtained. So these are two completely different things. We should therefore add a separate `restricted` value here.